### ISSUE TITLE
ModelCheckpoint does not check the value to monitor from logs modified with custom callback 

### ISSUE THREAD

**Issue by BishmoyPaul** [tensorflow/tensorflow#49500]
I created a custom callback that adds an additional metrics to the logs after each epoch during training. In a Modelcheckpoint callback, I put the additional metric to be monitored. But the Modelcheckpoint does not save according to the added metrics.
Sample code:
```
class LossCallback(tf.keras.callbacks.Callback):
    def on_epoch_end(self, epoch, logs=None):
        logs['new_loss']=logs['loss']+logs['val_loss']

save = tf.keras.callbacks.ModelCheckpoint(
        'model.h5', monitor='new_loss', verbose=1, save_best_only=True,
        save_weights_only=True, mode='min', save_freq='epoch')

model.fit(
       data, 
        epochs=10, 
        callbacks = [LossCallback(),save], 
        steps_per_epoch=100,
        validation_data=val_data, 
    )
```
Is there any way to get ModelCheckpoint to access the updated logs and save accordingly?

**Comment by UsharaniPagadala** [comment]
@BishmoyPaul 
We see that the issue template has not been filled, could you please do so as it helps us analyse the issue [tf version, steps followed before you ran into this error or stand alone code/cloab gist to reproduce the issue faced].
 Thanks!

**Comment by BishmoyPaul** [comment]
Issue fixed after adding `self._supports_tf_logs = True` after initializing the callback. Sample code given below :
```
class LossCallback(tf.keras.callbacks.Callback):
    def __init__(self):
        super(LossCallback, self).__init__()
        self._supports_tf_logs = True  ## Adding this fixed my issue

    def on_epoch_end(self, epoch, logs=None):
        logs['new_loss']=logs['loss']+logs['val_loss']
```
